packages/block-editor/src/components/rich-text/index.js

```diff
@@ -373,6 +373,8 @@ export class RichText extends Component {
 		// Browsers setting `isComposing` to `true` will usually emit a final
 		// `input` event when the characters are composed.
 		if ( event && event.nativeEvent.isComposing ) {
+			// Also don't update any selection.
+			document.removeEventListener( 'selectionchange', this.onSelectionChange );
```

Comment:
Do you think there's a simpler way togo instead of this selection enabling/disabling? maybe a flag?

Reply:
I see this as the simplest way. There's no need to run a listener if it's not needed?

Reply:
I mean, I think it's "cleanliest" if you remove listeners if you don't need them, add them back if you do need them again. I made the mistake before not to do that, and it had a negative performance impact (#12480), so I make a habit of taking this "cleaner" approach.